decimal.go

```diff
@@ -800,6 +812,30 @@ func (nd *NullDecimal) Scan(value interface{}) error {
 	return nd.Decimal.Scan(value)
 }
 
+var jsonNull = []byte("null")
+
+// UnmarshalText implements the encoding.TextUnmarshaler interface.
+func (nd *NullDecimal) UnmarshalText(b []byte) error {
+	if bytes.Compare(b, jsonNull) == 0 {
+		nd.Valid = false
+		return nil
+	}
+	if err := nd.Decimal.UnmarshalText(b); err != nil {
+		nd.Valid = false
+		return err
+	}
+	nd.Valid = true
+	return nil
+}
+
+// MarshalText implements the encoding.TextMarshaler interface.
+func (nd *NullDecimal) MarshalText() ([]byte, error) {
+	if nd.Valid {
+		return nd.Decimal.MarshalText()
+	}
+	return []byte("null"), nil
```

Comment:
@mjibson It's unclear to me what the proper string to use for marshaling null decimals.

We can support multiple values when unmarshling, but IDK of a clean way to support configurable marshling of multiple null values since the encoding.TextMarshaler doesn't provide one.